### ISSUE TITLE
restart cortex ruler not load rule config

### ISSUE THREAD

**Issue by AllenzhLi** [cortexproject/cortex#3008]
I use configdb to make a http request to get rule group config, when first run cortex ruler, it will store user rule group and rules to ./rules/username/filename, but when i restart cortex ruler with no any update, it will get rule group from configdb, and then mapper to ./rules/username/filename, ruler check there is no any update, just continue to boot. As a result, it ignore load any rules...

In short, 
1. ruler first start and get rules from configdb which make a http request
2. ruler set user rules to userManagers and mapping rules to a file
3. restart ruler
4. ruler second get fules form configdb
5. ruler compare and mapping rules to file
6. no update, then ruler alse not load to userManager
7. as a result, no any rules load

**Comment by AllenzhLi** [comment]
paste the func
```
// syncManager maps the rule files to disk, detects any changes and will create/update the
// the users Prometheus Rules Manager.
func (r *Ruler) syncManager(ctx context.Context, user string, groups store.RuleGroupList) {
	// A lock is taken to ensure if syncManager is called concurrently, that each call
	// returns after the call map files and check for updates
	r.userManagerMtx.Lock()
	defer r.userManagerMtx.Unlock()

	// Map the files to disk and return the file names to be passed to the users manager if they
	// have been updated
	update, files, err := r.mapper.MapRules(user, groups.Formatted())
	if err != nil {
		level.Error(r.logger).Log("msg", "unable to map rule files", "user", user, "err", err)
		return
	}

	if update {
		level.Debug(r.logger).Log("msg", "updating rules", "user", "user")
		configUpdatesTotal.WithLabelValues(user).Inc()
		manager, exists := r.userManagers[user]
		if !exists {
			manager, err = r.newManager(ctx, user)
			if err != nil {
				configUpdateFailuresTotal.WithLabelValues(user, "rule-manager-creation-failure").Inc()
				level.Error(r.logger).Log("msg", "unable to create rule manager", "user", user, "err", err)
				return
			}
			// manager.Run() starts running the manager and blocks until Stop() is called.
			// Hence run it as another goroutine.
			go manager.Run()
			r.userManagers[user] = manager
		}
		err = manager.Update(r.cfg.EvaluationInterval, files, nil)
		if err != nil {
			configUpdateFailuresTotal.WithLabelValues(user, "rules-update-failure").Inc()
			level.Error(r.logger).Log("msg", "unable to update rule manager", "user", user, "err", err)
			return
		}
	}
}
```

**Comment by pracucci** [comment]
@jtlisi Could you take a look, please? Is it something we're aware of?

**Comment by jtlisi** [comment]
This is an issue. We don't run into it internally because we run our rulers with the temporary directory pointed at an ephemeral disk. Anyone running on bare-metal will run into this.
